src/integrationTest/java/com/mongodb/spark/sql/connector/mongodb/MongoSparkConnectorTestCase.java

```diff
@@ -176,11 +177,20 @@ private void assertCommentsInProfile(
     MongoCollection<Document> profileCollection = database.getCollection("system.profile");
     try {
       profileCollection.drop();
-      database.runCommand(BsonDocument.parse(
-          format("{profile: 2, filter: {ns: '%s.%s'}}", databaseName, collectionName)));
+      database.runCommand(new Document("profile", 1)
+          .append(
+              "filter",
+              Filters.eq("ns", new MongoNamespace(databaseName, collectionName).getFullName())));
 
       runnable.run();
 
+      List<Document> profileDocsWithComment = profileCollection
+          .find(Filters.eq("command.comment", TEST_COMMENT))
+          .into(new ArrayList<>());
+      assertFalse(
+          profileDocsWithComment.isEmpty(),
+          format("There were no commands observed with the comment \"%s\"", TEST_COMMENT));
```

Comment:
Here we profile less, and we assert that we still observe commands with the `TEST_COMMENT`. I.e., this change makes the test stricter, it makes it actually test what is supposed to be tested.